homeassistant/components/cover/tilt.py

```diff
@@ -0,0 +1,156 @@
+    Tilt sensor and (toggle) switch to control a garage door opener or
+    motorized gate.
+    """
+
+    # pylint: disable=too-many-instance-attributes,too-many-arguments
+    def __init__(self, hass, dev_name, tilt_sensor_id, switch_id,
+                 contact_delay=DEFAULT_CONTACT_DELAY,
+                 run_time=DEFAULT_RUN_TIME):
+        """init the class"""
+        self._name = dev_name
+        self._hass = hass
+        self._tilt_sensor = tilt_sensor_id
+        self._switch = switch_id
+        self._contact_delay = contact_delay
+        self._run_time = run_time
+        self._state = STATE_OFF  # Assume closed
+        self._running = False  # is door in the process of opening or closing
+        self._delay_timer = None
+        self._run_timer = None
+
+    @property
+    def name(self):
+        """Return sensor name."""
+        return self._name
+
+    @property
+    def is_closed(self):
+        return self._state == STATE_CLOSED
+
+    def open_cover(self, **kwargs):
+        """Open garage door."""
+        _LOGGER.debug('tilt trying to open %s', self._name)
+        if self.is_closed:
+            _LOGGER.debug('tilt opening %s', self._name)
+            self._toggle()
+        self._state = STATE_OPEN
+        self.schedule_update_ha_state()
+
+    def close_cover(self, **kwargs):
+        """Close garage door."""
+        _LOGGER.debug('tilt trying to close %s', self._name)
+        if not self.is_closed:
+            _LOGGER.debug('closing %s', self._name)
+            self._toggle()
+        self._state = STATE_CLOSED
+        self.schedule_update_ha_state()
+
+    def _insure_relay_is_off(self):
+        """
+        Just in case the relay is stuck on (closed) turn it off.
+        If the relay is stuck in the on (closed) position then the garage
+        door opener will not work manually or via remote.
+        """
+        switch.turn_off(self.hass, self._switch)
+
+    def _toggle(self):
+        """Simulate a momentary contact button press."""
+
+        def _stop_delay_timer():
+            """Open momentary relay switch"""
+            switch.turn_off(self.hass, self._switch)
+
+        def _stop_run_timer():
+            """The door should be open/closed by now"""
+            _LOGGER.debug('tilt switch %s stopping timer', self._name)
+            self._running = False
+            self._insure_relay_is_off()
+
+        _LOGGER.debug("switch.turn_on %s", self._switch)
+        switch.turn_on(self.hass, self._switch)
+        self._running = True
+        self._delay_timer = Timer(self._contact_delay, _stop_delay_timer)
+        self._delay_timer.start()
+        self._run_timer = Timer(self._run_time, _stop_run_timer)
+        self._run_timer.start()
+
+    @property
+    def state(self):
+        return self._state
+
+    def update(self):
+        """Copy state of tilt sensor state of device."""
+        if self._running:
+            _LOGGER.debug('tilt switch %s running, assumed state is %s',
```

Comment:
All debug messages in t`update` seems useful for development but not for production.